android/src/main/kotlin/com/example/pocketape/ARCoreManager.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.pocketape
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.widget.Toast
+import androidx.core.content.ContextCompat
+import com.google.ar.core.*
+import com.google.ar.core.exceptions.*
+import com.google.ar.sceneform.ArSceneView
+
+class ARCoreManager(private val context: Context,  private val plugin: PocketapePlugin) {
+    private var session: Session? = null
+    private lateinit var arSceneView: ArSceneView
+    private var currentPosition: FloatArray? = null
+
+    init {
+        if (ContextCompat.checkSelfPermission(context, android.Manifest.permission.CAMERA)
+            == PackageManager.PERMISSION_GRANTED) {
+            initializeARSession()
+            initializeARSceneView()
+        }
+        else {
+            Toast.makeText(context, "Camera permission is required for AR functionality.", Toast.LENGTH_LONG).show()
+        }
```

Comment:
It is not possible to do it from the plugin since its context is not an Activity, I tried and It does not work. But we can move the configuration of the session in the Start method and it that case, as I mentioned before, it will slow down the start of the measurement.